atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -202,10 +202,13 @@
 
     abstract AtlasDbConfig config();
 
-    @Value.Default
-    Supplier<Optional<AtlasDbRuntimeConfig>> runtimeConfigSupplier() {
-        return Optional::empty;
-    }
+    abstract Optional<Refreshable<Optional<AtlasDbRuntimeConfig>>> runtimeConfig();
+
+    /**
+     * @deprecated use {@link #runtimeConfig} instead.
+     */
+    @Deprecated
+    abstract Optional<Supplier<Optional<AtlasDbRuntimeConfig>>> runtimeConfigSupplier();
```

Comment:
This change should be fine, the autogenerated immutables class should still have the same #runtimeConfigSupplier(Supplier) method, so should not be a break for users.

Reply:
See test in TransactionManagers I added, it's using this method.

Reply:
I guess my bigger concern is if someone calls `txMgr.runtimeConfigSupplier()` though a code search suggests that no one does so. 

If someone did (especially if that someone is one of our friendly supporting libs) that would be a spicy break 🔥but thankfully looks like that's not the case.

Reply:
Hmmm, I don't think runtimeConfigSupplier() would've compiled before? immutables shouldn't generate it, it only generates them when the type is Optional of something.